src/etwlogger_driver.c

```diff
@@ -218,7 +224,14 @@ void etwlogger_log(LOG_CATEGORY log_category, const char* file, const char* func
                 if (event_write_result != ERROR_SUCCESS)
                 {
                     /*fallback on printf...*/
-                    (void)printf("failed in EventWriteLogInfoEvent: %lu\r\n", event_write_result);
+                    if (InterlockedCompareExchange(&stopPrintfing, 1, 0) == 0)
+                    {
+                        (void)printf("failed in EventWriteLogInfoEvent: %lu. Further failing calls to EventWriteLogInfoEvent will not result in printf\r\n", event_write_result);
```

Comment:
>printf("failed in EventWriteLogInfoEvent: %lu. Further failing calls to EventWriteLogInfoEvent will not result in printf\r\n", event_write_result); [](start = 30, length = 147)

Do you really need to say you're going to stop printing, instead of just not printing?

Reply:
This is printed just once - and at that moment at least 1x ETW event is lost. Otherwise there would be an empty console that doesn't say anything and the user is left wondering "where's my events..."

---
In reply to: [319255960](https://github.com/Azure/azure-c-shared-utility/pull/365#discussion_r319255960) [](ancestors = 319255960)